File: BACKEND/MichaelPageBack/src/main/java/com/empresa/service/UsuarioService.java
package com.empresa.service;

import java.util.List;

import com.empresa.entity.Usuario;

public interface UsuarioService {

	List<String> listanombres();

	public abstract List<String> listaNombres();
	public abstract List<String> listaTelefonos();
	public abstract List<String> listaDirecciones();
	public abstract List<String> listaDepartamentos();
	public abstract List<String> listaProvincias(String departamento);
	public abstract List<Usuario> listaDistritos(String departamento,String provincia);
	
}

File: FRONTEND/MichaelPageFront/src/app/services/usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Usuario } from '../models/usuario.model';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})

export class UsuarioService {

  private apiServerURL = environment.apiBaseURL;

  constructor(private http: HttpClient) {}
  listarNombres(): Observable<string[]> {
    return this.http.get<string[]>(this.apiServerURL + '/rest/util/listaNombres');
  }

  listarTelefonos(): Observable<string[]> {
    return this.http.get<string[]>(this.apiServerURL + '/rest/util/listaTelefonos');
  }

  listarDirecciones(): Observable<string[]> {
    return this.http.get<string[]>(this.apiServerURL + '/rest/util/listaDirecciones');
  }

  listarDepartamento(): Observable<string[]> {
    return this.http.get<string[]>(this.apiServerURL + '/rest/util/listaDepartamentos');
  }

  listaProvincias(paramDep: any): Observable<string[]> {
    return this.http.get<string[]>(this.apiServerURL + '/rest/util/listaProvincias/' + paramDep);
  }

  listaDistritos(paramDep: any, paramProv: any): Observable<Usuario[]> {
    return this.http.get<Usuario[]>(this.apiServerURL + '/rest/util/listaDistritos/' + paramDep + '/' + paramProv);
  }
}
